RecoLocalCalo/CaloTowersCreator/src/CaloTowersCreationAlgo.cc

```diff
@@ -261,11 +265,15 @@ void CaloTowersCreationAlgo::setGeometry(const CaloTowerTopology* cttopo, const
   
   //which depths of tower 28/29 are merged?
   //the merging starts at layer 5 in phase 0 or phase 1 configurations
+  mergedDepths.clear();  mergedDepthsOne.clear();
   if(theHcalPhase==0 || theHcalPhase==1){
     std::vector<int> tower28depths;
     int ndepths, startdepth;
-    theHcalTopology->getDepthSegmentation(theHcalTopology->lastHERing()-1,tower28depths);
-    theHcalTopology->depthBinInformation(HcalEndcap,theHcalTopology->lastHERing()-1,1,1,ndepths,startdepth);
+    subdetOne = theHcalTopology->getPhiZOne(phizOne);
+    int zside = (subdetOne > 0) ? -phizOne[0].second : 1;
```

Comment:
this sign flip looks confusing. Is the intrinsic assumption that the other z side of the  "one" is with the normal segmentation?